src/main/webapp/js/auscope/widgets/GAOnlineResourcesPanel.js

```diff
@@ -0,0 +1,575 @@
+        var sortable = true;
+        var hideHeaders = true;
+        if (typeof(cfg.hideHeaders) !== 'undefined' && cfg.hideHeaders != null) {
+            hideHeaders = cfg.hideHeaders;
+        }
+        if (typeof(cfg.sortable) !== 'undefined' && cfg.sortable != null) {
+            sortable = cfg.sortable;
+        }
+
+        //We allow the owner to specify additional columns
+        var columns = [{
+            //Title column
+            dataIndex: 'onlineResource',
+            menuDisabled: true,
+            sortable: sortable,
+            width: 480,
+            cellWrap : true,
+            renderer: Ext.bind(me._detailsRenderer, me)
+        },
+        {
+            dataIndex: 'onlineResource',
+            width: 170,
+            renderer: Ext.bind(me._linksRenderer, me)
+        },
+        {
+            dataIndex: 'onlineResource',
+            width: 150,
+            renderer: Ext.bind(me._previewRenderer, me)
+        }];
+        if (cfg.columns) {
+            columns = columns.concat(cfg.columns);
+        }
+        
+        //Build our configuration object
+        Ext.apply(cfg, {
+            selModel: cfg.selModel,
+            features : [groupingFeature],
+            store : Ext.create('Ext.data.Store', {
+                groupField : 'group',
+                model : 'portal.widgets.panel.OnlineResourcePanelRow',
+                data : dataItems
+            }),
+            plugins : [{
+                ptype : 'selectablegrid'
+            }],
+            hideHeaders : hideHeaders,
+            columns: columns,
+            viewConfig: {
+                enableTextSelection: true,
+                listeners: {
+                   
+                    // when the view is fully loaded we need to check for availability of some features and update the DOM                    
+                    viewready: function(view) {
+                        for (var i = 0; i < cfg.cswRecords.length; i++) {
+                            var onlineResources = cfg.cswRecords[i].data.onlineResources;
+                            for (var j = 0; j < onlineResources.length; j++) {
+                                var onlineResource = onlineResources[j]; 
+                                if (onlineResource.get('type') == portal.csw.OnlineResource.WFS
+                                        || onlineResource.get('type') == portal.csw.OnlineResource.WMS) {
+                                    var serviceUrl = onlineResource.get('url');
+                                    var version = onlineResource.get('version');  
+                                    var name = onlineResource.get('name');                                
+                                    var id = onlineResource.get('id');                                    
+                                    var description = onlineResource.get('description');   
+                                    auscope.widgets.GAOnlineResourcesPanel.prototype.

```

Comment:
Technical debt - I don't believe this is the correct way of doing this.  That is, prototype defines a class definition whereas I'm sure you want to use the instance.  Then again I may be wrong.